tests/unit/string_lib_test.c

```diff
@@ -496,6 +496,42 @@ static void test_string_to_long_compatibility(void)
     // Old function (StringToLongUnsafe) does not allow trailing whitespace
 }
 
+static void test_string_decimal_to_long(void)
+{
+   // Basic usage:
+   assert_int_equal(StringDecimalToLong("0.0"), 0);
+   assert_int_equal(StringDecimalToLong("-0.0"), 0);
+   assert_int_equal(StringDecimalToLong("+0.0"), 0);
+   assert_int_equal(StringDecimalToLong("123.0"), 123);
+   assert_int_equal(StringDecimalToLong("+123.0"), 123);
+   assert_int_equal(StringDecimalToLong("-123.0"), -123);
+
+   // Leading spaces:
+   assert_int_equals(StringDecimalToLong(" 0.0"), 0);
+   assert_int_equals(StringDecimalToLong(" 123.0"), 123);
+   assert_int_equals(StringDecimalToLong(" -123.0"), -123);
+   assert_int_equals(StringDecimalToLong("           0.0"), 0);
+   assert_int_equals(StringDecimalToLong("           123.0"), 123);
+   assert_int_equals(StringDecimalToLong("           -123.0"), -123);
+
+   // Trailing spaces:
+   assert_int_equals(StringDecimalToLong("9.0 "), 9);
+   assert_int_equals(StringDecimalToLong("789.0 "), 789);
+   assert_int_equals(StringDecimalToLong("-789.0 "), -789);
+   assert_int_equals(StringDecimalToLong("9.0           "), 9);
+   assert_int_equals(StringDecimalToLong("789.0         "), 789);
+   assert_int_equals(StringDecimalToLong("-789.0        "), -789);
+
+   // More spaces:
+   assert_int_equals(StringDecimalToLong("   0.0     "), 0);
+   assert_int_equals(StringDecimalToLong("   -0.0    "), 0);
+   assert_int_equals(StringDecimalToLong("   123.0  "), 123);
+
+   // Space separated numbers:
+   assert_int_equals(StringDecimalToLong("   123.456 78   9  "), 123);
+   assert_int_equals(StringDecimalToLong("123.0 45"), 123);
```

Comment:
You should add a test case for testing the error handling as well. Send in a too long number and check that it returns nonzero and also check that the string error message is correct.

Reply:
How can i check that the string logged is correct? it is immidiatelly passed to `Log`

Reply:
Ah, I didn't realize. Then it's not so easy, so drop it.